cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -580,6 +581,18 @@ def set_final_status():
         hookenv.status_set('active', msg)
         return
 
+    if is_state('ceph-storage.available') and \
+            is_state('ceph-storage.available') and \
+            is_state('kubernetes-master.privileged') and \
+            not is_state('kubernetes-master.ceph.configured'):
+
+        ceph_admin = endpoint_from_flag('ceph-storage.available')
+
+        if get_version('kube-apiserver') >= (1, 12) and \
+                not ceph_admin.key():
+            hookenv.status_set(
+                'waiting', 'Waiting for Ceph to provide a key.')
```

Comment:
You need an early `return` here, otherwise this will get overwritten by the `Kubernetes master running` status below.